server/v2/commands.go

```diff
@@ -65,12 +84,55 @@ func Commands(logger log.Logger, homePath string, modules ...ServerModule) (CLIC
 	return cmds, nil
 }
 
-func AddCommands(rootCmd *cobra.Command, logger log.Logger, homePath string, modules ...ServerModule) error {
-	cmds, err := Commands(logger, homePath, modules...)
+func AddCommands(rootCmd *cobra.Command, newApp AppCreator[transaction.Tx], logger log.Logger, components ...ServerComponent[transaction.Tx]) error {
+	cmds, err := Commands(rootCmd, newApp, logger, components...)
 	if err != nil {
 		return err
 	}
 
+	server := NewServer(logger, components...)
+	originalPersistentPreRunE := rootCmd.PersistentPreRunE
+	rootCmd.PersistentPreRunE = func(cmd *cobra.Command, args []string) error {
```

Comment:
let's check if `PersistentPreRun` is set as well (it could be that a chain used that instead of PersistentPreRunE), set it below and set it to nil afterwards (as we want to use PersistentPreRunE)